src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/casenotes/CaseNotesSynchronisationService.kt

```diff
@@ -39,13 +40,21 @@ class CaseNotesSynchronisationService(
   }
 
   suspend fun caseNoteInserted(event: CaseNotesEvent) {
-    val nomisCaseNote = nomisApiService.getCaseNote(event.caseNoteId)
-    if (nomisCaseNote.isSourcedFromDPS()) {
-      telemetryClient.trackEvent("casenotes-synchronisation-created-skipped", event.toTelemetryProperties())
-      return
-    }
-
     try {
+      val nomisCaseNote = nomisApiService.getCaseNote(event.caseNoteId)
+      if (nomisCaseNote.isSourcedFromDPS()) {
+        telemetryClient.trackEvent("casenotes-synchronisation-created-skipped", event.toTelemetryProperties())
+        return
+      }
+      if (nomisCaseNote.auditMissing()) {
+        val cn = caseNotesMappingService.getMappingGivenNomisIdOrNull(event.caseNoteId)
+        if (cn != null && cn.mappingType == DPS_CREATED) {
+          // Detected where the auditModuleName is null but the mapping exists, signifying that this CN was created from DPS
+          telemetryClient.trackEvent("casenotes-synchronisation-created-skipped-null", event.toTelemetryProperties())
+          return
+        }
+      }
```

Comment:
Code added just specifically for the case where audit is null, the mapping already exists and was originally created by DPS.